docs/week-3.md

```diff
@@ -0,0 +1,568 @@
+import React from 'react'; 
+
+const LuckyNum = () => {
+
+    let myLuckyNumber = 7;
+
+    return (
+        <div>
+            <h1>My Lucky Number is {myLuckyNumber}</h1>
+        </div>
+    );
+}
+
+export default LuckyNum;
+```
+
+Right now this component only displays John's lucky number. In order to make this component more reusable so that
+Doug can use it, we must add props. We add props as a parameter of the function component.
+
+```
+import React from 'react'; 
+
+const LuckyNum = (props) => {
+
+    let myLuckyNumber = 7;
+
+    return (
+        <div>
+            <h1>My Lucky Number is {myLuckyNumber}</h1>
+        </div>
+    );
+}
+
+export default LuckyNum;
+```
+
+Now that we have access to `props` we can simply display the `luckyNum` field of the prop object.
+
+```
+import React from 'react'; 
+
+const LuckyNum = (props) => (
+
+    <div>
+        <h1>My Lucky Number is {props.luckyNumber}</h1>
+    </div>
+
+)
+
+export default LuckyNum;
+```
+Our component is now reusable!! We now can use it on Doug's profile like so.
+
+```
+import React from 'react';
+import LuckyNum from './LuckyNum';
+
+const Profile = () => (
+
+    <div>
+        <img src="profileImage.jpg"/>
+        <h1>Hi, my name is Doug! My favorite color is purple.</h1>
+        <LuckyNum luckyNum={1111}/>
+    </div>
+);
+
+export default Profile;
+```
+
+`props` is an object that React makes for each component. When you pass parameters into component declaration 
+like we did with `LuckyNum` React adds the parameter as a field of the object `props`. In order to get the field
+of the object we simply access the object's field like we would any other object. 
+
+> We can make the `Profile` component more reusable with props. Try now to add props to make `Profile` more
+>generic (possible answer below). 
+
+```
+import React from 'react';
+import LuckyNum from './LuckyNum';
+
+const Profile = (props) => (
+
+    <div>
+        <img src=`${props.profileURL}`/>
+        <h1>Hi, my name is {props.name}! My favorite color is {props.favoriteColor}.</h1>
+        <LuckyNum luckyNum={props.luckyNum}/>
+    </div>
+);
+
+export default Profile;
+```
+
+## State Management
+
+Alright. So John is now super happy with his profile and wants to see if other people like his profile as much
+as he does. To do this John adds a like button on his profile. 
+
+```
+import React from 'react';
+import LuckyNum from './LuckyNum';
+
+const Profile = () => (
+
+    <div>
+        <img src="profileImage.jpg"/>
+        <h1>Hi, my name is John! My favorite color is blue.</h1>
+        <LuckyNum />
+        <button>Likes: </button>
+    </div>
+);
+
+export default Profile;
+```
+
+John is now perplexed. How can I keep track of the number of likes and how can I keep the number consistent
+through each render? Welcome to state management John.
+
+### What is managing state?
+
+You can think of state as the data that persists through each clock render of your application. Under the hood,
```

Comment:
I'm not sure these two sentences are correct. React will only re-render when data changes. I think it would be more accurate to say the state is all the data that can change while a user is on a webpage.